database/src/main/kotlin/no/nav/su/se/bakover/database/RevurderingPostgresRepo.kt

```diff
@@ -0,0 +1,178 @@
+                id = id,
+                periode = periode,
+                opprettet = opprettet,
+                tilRevurdering = tilRevurdering, beregning = beregning, saksbehandler = Saksbehandler(saksbehandler)
+            )
+            else -> OpprettetRevurdering(
+                id = id,
+                periode = periode,
+                opprettet = opprettet,
+                tilRevurdering = tilRevurdering,
+                saksbehandler = Saksbehandler(saksbehandler)
+            )
+        }
+    }
+
+    private fun lagre(revurdering: OpprettetRevurdering) =
+        dataSource.withSession { session ->
+            (
+                "insert into revurdering" +
+                    "(id, opprettet, behandlingId, periode, beregning, simulering, saksbehandler, oppgaveId)" +
+                    "values" +
+                    "(:id, :opprettet, :behandlingId, to_json(:periode::json), null, null, :saksbehandler, :oppgaveId)" +
+                    "on conflict (id) do " +
+                    "update set beregning = null, simulering = null"
+                ).oppdatering(
+                mapOf(
+                    "id" to revurdering.id,
+                    "periode" to objectMapper.writeValueAsString(revurdering.periode),
+                    "opprettet" to revurdering.opprettet,
+                    "behandlingId" to revurdering.tilRevurdering.id,
+                    "saksbehandler" to revurdering.saksbehandler.navIdent
+                ),
+                session
+            )
+        }
+
+    private fun lagre(revurdering: BeregnetRevurdering) =
+        dataSource.withSession { session ->
+            (
+                "insert into revurdering" +
+                    "(id, opprettet, behandlingId, periode, beregning, simulering, saksbehandler, oppgaveId)" +
+                    "values" +
+                    "(:id, :opprettet, :behandlingId, to_json(:periode::json), to_json(:beregning::json), null, :saksbehandler, :oppgaveId)" +
+                    "on conflict (id) do " +
+                    "update set beregning = to_json(:beregning::json), simulering = null"
+                ).oppdatering(
+                mapOf(
+                    "id" to revurdering.id,
+                    "periode" to objectMapper.writeValueAsString(revurdering.periode),
+                    "opprettet" to revurdering.opprettet,
+                    "behandlingId" to revurdering.tilRevurdering.id,
+                    "beregning" to objectMapper.writeValueAsString(revurdering.beregning),
+                    "saksbehandler" to revurdering.saksbehandler.navIdent
+                ),
+                session
+            )
+        }
+
+    private fun lagre(revurdering: SimulertRevurdering) =
+        dataSource.withSession { session ->
+            (
+                "insert into revurdering" +
+                    "(id, opprettet, behandlingId, periode, beregning, simulering, saksbehandler, oppgaveId)" +
```

Comment:
For behandling har vi fjernet insert fra de plassene der det er unaturlig å gjøre en insert. Vi beholdt den kun for "opprett", som vi har renamet til Vilkårsvurdert.Uavklart